### PR TITLE
[Port] Telecommunications Updated Machine UI

### PR DESCRIPTION
**Pull request by Sarchutar**
## About The Pull Request

Port of: https://github.com/tgstation/tgstation/pull/52357

Makes all telecomms machines use tgui.
I additionally added the list for all text to radio circuits back to the receiver interface and made some small fixes/changes to some things (check commit messages for this).

If there is any follow up PR or anything missing let me know.

<details>
<summary>
Picture
</summary>

tcomms_tgui

</details>

## Why It's Good For The Game

HTML just hurts my eyes and this just looks better.

## Changelog
:cl: WondaMegapon, Sarchutar
add: Telecommunication machines now use a TGUI based interface
del: Removed HTML based telecommunication machine code
balance: Telecommunication machines can have their interfaces viewed without a multitool
server: Telecommunication machines now log changed settings
/:cl:

### CODE REVIEW

**Comment by Mat05usz** [comment]
Based

**Comment by francinum** [comment]
I have no idea why these need to be TGUI, but sure whatever

**Comment by francinum** [comment]
At least set the theme to `hackerman`

**Comment by Sarchutar** [comment]
> At least set the theme to `hackerman`

Are you sure about that? This just makes it look worse.

hackerman

**Comment by francinum** [comment]
When the hell did hackerman get beveled buttons...
Yeah in that case the standard theme is fine.

**Comment by Sarchutar** [comment]
I will resolve that conflict later.

**Comment by Sarchutar** [comment]
Alright, lets hope i resolved that correctly since i never resolved tgui conflicts before.

File: code/game/machinery/telecomms/machine_interactions.dm [hunk 1]
```diff
@@ -27,74 +29,202 @@
...
+		var/obj/item/integrated_circuit/I = c
+		var/obj/item/O = I.get_object()
+		var/list/circuit = list()
+		if(get_area(O)) //if it isn't in nullspace, can happen due to printer newing all possible circuits to fetch list data
+			n++
+			circuit["index"] = n
+			circuit["name"] = O.name
+			circuit["coords"] = "[O.x], [O.y], [O.z]"
+			circuits += list(circuit)
+	data["circuits"] = circuits
```
Comment (park66665) [inline code/game/machinery/telecomms/machine_interactions.dm]: I think you can move this portion to `/obj/machinery/telecomms/receiver/add_option()`

File: code/game/machinery/telecomms/machine_interactions.dm [hunk 2]
```diff
@@ -27,74 +29,202 @@
...
+
+/obj/machinery/telecomms/ui_act(action, params)
+	if(..())
+		return
 
-		dat += "<br>  <a href='?src=[REF(src)];input=freq'>\[Add Filter\]</a>"
-		dat += "<hr>"
+	if(!issilicon(usr))
+		if(!istype(usr.get_active_held_item(), /obj/item/multitool))
+			return
```
Comment (park66665) [inline code/game/machinery/telecomms/machine_interactions.dm]:
```suggestion
	if(!issilicon(operator) && !istype(operator.get_active_held_item(), /obj/item/multitool))
		return
```

Using `usr` (it really is typed as `/mob`) here is mostly okay, but let's be safe and consistent.

(Speaking of consistency, you probably *can* replace `operator` with `usr` if you *really* want.)

File: code/game/machinery/telecomms/machine_interactions.dm [hunk 3]
```diff
@@ -27,74 +29,202 @@
...
+/obj/machinery/telecomms/bus/add_act(action, params)
+	switch(action)
+		if("change_freq")
+			var/newfreq = text2num(params["value"]) * 10
+			if(newfreq)
+				if(newfreq < 10000)
+					change_frequency = newfreq
+					. = TRUE
+				else
+					change_frequency = 0
```
Comment (park66665) [inline code/game/machinery/telecomms/machine_interactions.dm:223]:
```suggestion
					change_frequency = 0
					. = TRUE
```

File: code/game/machinery/telecomms/machine_interactions.dm [hunk 4]
```diff
@@ -27,74 +29,202 @@
...
+						. = TRUE
+		if("buffer")
+			heldmultitool.buffer = src
+			. = TRUE
+		if("flush")
+			heldmultitool.buffer = null
+			. = TRUE
+
+	add_act(action, params)
+	. = TRUE
```
Comment (park66665) [inline code/game/machinery/telecomms/machine_interactions.dm]:
```suggestion
	if(add_act(action, params))
		. = TRUE
```

**Review by park66665 — COMMENTED** [review]
I took a cursory look at the code. Things are generally fine, and the UIs work. (well, it shows up and buttons work)

**Review by park66665 — APPROVED** [review]
All clear. This PR is good to go.

**Pull request merged**